src/mediaelement.js

```diff
@@ -60,6 +60,43 @@ WaveSurfer.util.extend(WaveSurfer.MediaElement, {
         this.setPlaybackRate(this.playbackRate);
     },
 
+    loadElt: function (elt, container, peaks) {
+        var my = this;
+
+        var media = elt;
+        media.controls = this.params.mediaControls;
+        media.autoplay = this.params.autoplay || false;
+        media.style.width = '100%';
+
+        media.addEventListener('error', function () {
+            my.fireEvent('error', 'Error loading media element');
+        });
+
+        media.addEventListener('canplay', function () {
+            my.fireEvent('canplay');
+        });
+
+        media.addEventListener('ended', function () {
+            my.fireEvent('finish');
+        });
+
+        media.addEventListener('timeupdate', function () {
+            my.fireEvent('audioprocess', my.getCurrentTime());
+        });
+
+        var prevMedia = container.querySelector(this.mediaType);
+        if (prevMedia) {
+            container.removeChild(prevMedia);
+        }
+        container.appendChild(media);
```

Comment:
Just to clarify: this will move the existing media element from its initial position in DOM to the wavesurfer.js' container. Are you sure we want this?